src/angular-gs-flash-message.js

```diff
@@ -23,13 +23,19 @@ angular.module('gs.flash-message', ['ngSanitize'])
     restrict: 'A',
     scope: true,
     template:
-      '<div class="alert" ng-if="messages.error.length || messages.success.length">' +
+      '<div class="alert" ng-if="messages.error.length || messages.success.length || messages.warning.length">' +
         '<div ng-if="messages.error.length" class="alert--message alert--message__error">' +
           '<span class="alert--message--content" ng-repeat="error in messages.error track by $index" ng-bind-html="error"></span>' +
           '<a href="javascript:;" ng-click="close()">' +
             '<i class="alert--message--remove-icon"></i>' +
           '</a>' +
         '</div>' +
+        '<div ng-if="messages.warning.length" class="alert--message alert--message__warning">' +
```

Comment:
Do you mean iterating over the object keys and interpolating those keys into the message fetching and CSS classes?